sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -755,8 +563,16 @@ def upload_blob(
                 :dedent: 12
                 :caption: Upload a blob to the container.
         """
-        options = self._upload_blob_options(
-            data,
+        options = _upload_blob_options(
+            require_encryption=self.require_encryption,
+            encryption_version=self.encryption_version,
+            key_encryption_key=self.key_encryption_key,
+            key_resolver_function=self.key_resolver_function,
```

Comment:
I would ideally like to reduce all these things we have to pass but it may be a little tricky. Can we at least package these into `encryption_options` here before passing rather than inside the function and just pass that. Might be fun to type that way but would be a little nicer. Will require minor changes to the helper function. 

https://github.com/Azure/azure-sdk-for-python/blob/ca79216ff9fe86b1a1b97166ec4986f4a29d572f/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py#L380-L385

Perhaps we even build and store this object on class construction?